api/v1alpha1/contour_types.go

```diff
@@ -229,6 +229,41 @@ type EnvoyNetworkPublishing struct {
 	// +kubebuilder:validation:MaxItems=2
 	// +kubebuilder:default={{name: http, portNumber: 8080}, {name: https, portNumber: 8443}}
 	ContainerPorts []ContainerPort `json:"containerPorts,omitempty"`
+
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +optional
+	Tolerations []Toleration `json:"tolerations,omitempty"`
+}
+
+type Toleration struct {
```

Comment:
any particular reason to use a custom type here, vs. reusing the upstream `corev1.Toleration`?

Reply:
I prefer using `corev1.Toleration` instead of supporting a custom type.